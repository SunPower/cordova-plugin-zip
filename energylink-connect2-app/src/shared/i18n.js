import { useMemo } from 'react'
import { useSelector } from 'react-redux'

import * as languages from './locales'
import * as localizedComponents from './localizedComponents'

const getTranslate = translations => (key, ...params) => {
  let value = translations[key] || key
  if (params) {
    params.forEach((p, idx) => {
      value = value.replace(`{${idx}}`, p)
    })
  }
  return value
}

export const useI18n = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language
  return useMemo(() => getTranslate(languages[locale]), [locale])
}

export const useI18nComponent = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language || {}

  return useMemo(
    () => componentName => {
      const path = `${locale}_${componentName}`
      const Component = localizedComponents[path] || {}
      if (typeof Component !== 'function') {
        throw new Error(`Component not found: ${path}`)
      }
      return Component()
    },
    [locale]
  )
}
