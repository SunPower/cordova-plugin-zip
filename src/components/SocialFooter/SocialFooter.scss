@import "../../styles/variables";

.sp-social-share {
  flex-direction: column;
  justify-content: center;

  .white-text {
    color: $gray-text-alt;
  }

  .dark-text {
    color: $black;
  }

  h3 {
    text-align: center;
    font-weight: $bold-text;
    font-size: $font-size;
    line-height: 1rem;
    text-align: center;
    letter-spacing: $letter-spacing-big;
    margin-bottom: 0.5rem;
    text-transform: uppercase;
  }

  .share-icons {
    justify-content: space-between;
    height: 3rem;
    align-content: center;
    align-items: center;
    margin-bottom: 1rem;

    svg {
      cursor: pointer;
    }
  }
}
