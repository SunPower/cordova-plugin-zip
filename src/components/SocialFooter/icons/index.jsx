export { ReactComponent as FacebookIcon } from './facebook-icon.svg'
export { ReactComponent as TwitterIcon } from './twitter-icon.svg'
export { ReactComponent as LinkedinIcon } from './linkedin-icon.svg'
export { ReactComponent as YoutubeIcon } from './youtube-icon.svg'
export { ReactComponent as InstagramIcon } from './instagram-icon.svg'
