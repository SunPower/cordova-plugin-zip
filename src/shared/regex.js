export const PASSWORD_REGEXP = /^(?=.*[a-z])(?=.*\d)[A-Za-z\d]{8,}$/
export const EMAIL_REGEXP = /^\S+@\S+\.\S+$/
export const PHONE_REGEXP = /^(1\s?)?((([0-9]{3}))|[0-9]{3})[\s-]?[0-9]{3}[\s-]?[0-9]{4}$/
