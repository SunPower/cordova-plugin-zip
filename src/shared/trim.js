export const trimObject = obj => {
  return Object.entries(obj)
    .map(([k, v]) => ({
      [k]: typeof v === 'string' || v instanceof String ? v.trim() : v
    }))
    .reduce((acc, kv) => ({ ...acc, ...kv }), {})
}
