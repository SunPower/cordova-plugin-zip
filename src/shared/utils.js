export const either = (condition, whenTrue, whenFalse) =>
  condition ? whenTrue : whenFalse
