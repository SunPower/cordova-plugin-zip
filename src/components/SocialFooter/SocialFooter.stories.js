import React from 'react'

import { storiesOf } from '@storybook/react'

import SocialFooter from './index'

storiesOf('SocialFooter', module).add('Simple', () => <SocialFooter />)
