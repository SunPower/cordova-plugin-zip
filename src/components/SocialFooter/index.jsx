import React from 'react'
import clsx from 'clsx'
import { useI18n } from '../../shared/i18n'
import {
  FacebookIcon,
  InstagramIcon,
  LinkedinIcon,
  TwitterIcon,
  YoutubeIcon
} from './icons'

import './SocialFooter.scss'

function SocialFooter({ invert = false }) {
  const {
    REACT_APP_FACEBOOK_URL,
    REACT_APP_TWITTER_URL,
    REACT_APP_LINKEDIN_URL,
    REACT_APP_YOUTUBE_URL,
    REACT_APP_INSTAGRAM_URL
  } = process.env
  const t = useI18n()
  const defaultTheme = invert ? 'white-text' : 'dark-text'
  const classNames = clsx(defaultTheme, 'pb-20')

  return (
    <div className="sp-social-share is-flex">
      <h3 className={classNames}>{t('SOCIAL_FOOTER_TITLE')}</h3>
      <div className="share-icons is-flex">
        <a
          target="_blank"
          href={REACT_APP_FACEBOOK_URL}
          rel="noopener noreferrer"
        >
          <FacebookIcon />
        </a>
        <a
          target="_blank"
          href={REACT_APP_TWITTER_URL}
          rel="noopener noreferrer"
          className="pt-5"
        >
          <TwitterIcon />
        </a>
        <a
          target="_blank"
          href={REACT_APP_LINKEDIN_URL}
          rel="noopener noreferrer"
        >
          <LinkedinIcon />
        </a>
        <a
          target="_blank"
          href={REACT_APP_YOUTUBE_URL}
          rel="noopener noreferrer"
          className="pt-5"
        >
          <YoutubeIcon />
        </a>
        <a
          target="_blank"
          href={REACT_APP_INSTAGRAM_URL}
          rel="noopener noreferrer"
          className="pt-3"
        >
          <InstagramIcon />
        </a>
      </div>
    </div>
  )
}

export default SocialFooter
