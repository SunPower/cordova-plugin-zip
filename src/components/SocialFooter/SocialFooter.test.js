import React from 'react'
import { shallow } from 'enzyme'
import SocialFooter from '.'
import * as i18n from '../../shared/i18n'

beforeEach(() => {
  process.env = {
    REACT_APP_EDP_API_URL: 'https://dev-elhapi.dev-edp.sunpower.com/v1/elh',
    REACT_APP_FACEBOOK_URL: 'https://www.facebook.com/sunpower',
    REACT_APP_TWITTER_URL: 'https://twitter.com/SunPower',
    REACT_APP_LINKEDIN_URL:
      'https://www.linkedin.com/company/sunpower-corporation',
    REACT_APP_YOUTUBE_URL: 'https://www.youtube.com/user/sunpower',
    REACT_APP_INSTAGRAM_URL: 'https://www.instagram.com/sunpower'
  }
})

describe('Social component', () => {
  it('renders without crashing', () => {
    jest
      .spyOn(i18n, 'useI18n')
      .mockImplementation(path => (key, ...params) =>
        `${key.toUpperCase()} ${params.join('_')}`.trim()
      )
    const component = shallow(<SocialFooter />)
    expect(component).toMatchSnapshot()
  })
})
