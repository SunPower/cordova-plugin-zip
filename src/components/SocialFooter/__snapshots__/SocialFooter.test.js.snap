// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social component renders without crashing 1`] = `
<div
  className="sp-social-share is-flex"
>
  <h3
    className="dark-text pb-20"
  >
    SOCIAL_FOOTER_TITLE
  </h3>
  <div
    className="share-icons is-flex"
  >
    <a
      href="https://www.facebook.com/sunpower"
      rel="noopener noreferrer"
      target="_blank"
    >
      <ForwardRef(SvgFacebookIcon) />
    </a>
    <a
      className="pt-5"
      href="https://twitter.com/SunPower"
      rel="noopener noreferrer"
      target="_blank"
    >
      <ForwardRef(SvgTwitterIcon) />
    </a>
    <a
      href="https://www.linkedin.com/company/sunpower-corporation"
      rel="noopener noreferrer"
      target="_blank"
    >
      <ForwardRef(SvgLinkedinIcon) />
    </a>
    <a
      className="pt-5"
      href="https://www.youtube.com/user/sunpower"
      rel="noopener noreferrer"
      target="_blank"
    >
      <ForwardRef(SvgYoutubeIcon) />
    </a>
    <a
      className="pt-3"
      href="https://www.instagram.com/sunpower"
      rel="noopener noreferrer"
      target="_blank"
    >
      <ForwardRef(SvgInstagramIcon) />
    </a>
  </div>
</div>
`;
